File: timer.cpp
#include<iostream>
#include<chrono>
#include<string>
typedef std::string string;
#include"hpcalc.h"
using namespace grnum;
int main() {
    int digits = 1e7;
    string s1(digits, '3');
    string s2(digits, '8');

    HP a = s1;
    HP b = s2;
    HP c;

    puts("预热中...");
        c = a + b;
//        c = a - b;
//        c = a * b;
//        c = a / b;

    puts("开始测试");
    const int TEST_COUNT = 5;
    long long times[TEST_COUNT];

    for(int t = 0; t < TEST_COUNT; t++) {
        auto st = std::chrono::steady_clock::now();
        // 被测运算
        c = a%998244353;
        // C = A - B;
        // C = A * B;
        // C = A / B;
        auto ed = std::chrono::steady_clock::now();
        times[t] = std::chrono::duration_cast<std::chrono::milliseconds>(ed - st).count();
        std::cout << "第" << t+1 << "次: " << times[t] << " ms" << std::endl;
    }

    // 计算平均
    long long sum = 0;
    for(int t = 0; t < TEST_COUNT; t++) sum += times[t];
    std::cout << "\n平均: " << sum / TEST_COUNT << " ms" << std::endl;

    return 0;
}

File: timer.py
import time
import sys

def test_python_int():
    # 准备数据
    digits = 10000000  # 和C++保持一致
    s1 = '7' * digits
    s2 = '5' * digits
    print("准备构造")
    sys.set_int_max_str_digits(1000000000)
    A = int(s1)
    B = int(s2)
    
    # 预热
    print("预热中...")
    C = A + B
    # C = A - B
    # C = A * B
    # C = A // B  # Python整数除法
    
    # 正式测试
    TEST_COUNT = 5
    times = []
    
    for t in range(TEST_COUNT):
        start = time.perf_counter()
        
        # 被测运算（和C++保持一致）
        C = A%998244353
        # C = A - B
        # C = A * B
        # C = A // B
        
        end = time.perf_counter()
        ms = (end - start) * 1000
        times.append(ms)
        print(f"第{t+1}次: {ms:.2f} ms")
    
    print(f"\n平均: {sum(times)/TEST_COUNT:.2f} ms")

if __name__ == "__main__":
    test_python_int()